### ISSUE TITLE
Setting output.globalObject breaks parsed + gzip 

### ISSUE THREAD

**Issue by ntucker** [webpack-contrib/webpack-bundle-analyzer#288]
#### Issue description
```js
config.output.globalObject = "(typeof self !== 'undefined' ? self : this)"
```
Breaks parsed + gzip sizes. Removing it fixed them.

#### Technical info
* Webpack Bundle Analyzer version: 3.3.2
* Webpack version: 4.35.0
* Node.js version: 11.10.0
* npm/yarn version: 6.7.0
* OS: mac os x; windows

### Expected Behavior

Gzip and parsed should contain data bout more than just the output files - the modules inside them as well.

### Actual Behavior

Only stats includes all modules. Gzip and parsed just show large blocks for each bundle piece (output file)

### Code

My default:

```js
config.output.globalObject = "(typeof self !== 'undefined' ? self : this)"
```

When I analyze:
```js
delete config.output.globalObject;
```

**Comment by valscion** [comment]
Sorry, I don't quite follow you. Is the piece of code you are referring to some part of `webpack-bundle-analyzer` or your own code?

**Comment by ntucker** [comment]
the config setting lines are setting the webpack config. This issue describes what happens to any bundle when the configuration is as such.

**Comment by valscion** [comment]
I'm having a hard time figuring out what is exactly what's happening here. Can you create a small reproduction repository to show what's happening?

I'm afraid I can't help you otherwise

**Comment by ntucker** [comment]
https://webpack.js.org/configuration/output/#outputglobalobject This is the webpack configuration. You can use with any project. Just set that to  "(typeof self !== 'undefined' ? self : this)" and add bundle analyzer

**Comment by valscion** [comment]
But why would you use `self` or `this` in there? Do you mean `'this'` instead of `this`?

**Comment by ntucker** [comment]
https://developer.mozilla.org/en-US/docs/Web/API/WorkerGlobalScope/self

**Comment by valscion** [comment]
Yes but the webpack configuration expects you to put a string in there, not an object, unless I misunderstand the instructions in https://webpack.js.org/configuration/output/#outputglobalobject documentation you linked?

**Comment by ntucker** [comment]
"(typeof self !== 'undefined' ? self : this)" is a string literal. In javascript this is signified by the quotation marks surrounding the other characters.

**Comment by valscion** [comment]
But it won't be evaluated as JS unless you put it in backticks? It means your string will be exactly what you write, not either `'self'` or `'this'`.

Was this what you have, or would need?

```js
config.output.globalObject = (typeof self !== 'undefined' ? 'self' : 'this')
```

This code will not do any conditionals:

```js
config.output.globalObject = "(typeof self !== 'undefined' ? self : this)"
```

The result will be only the raw string `"(typeof self !== 'undefined' ? self : this)"` as is, not `self` or `this`.

**Comment by th0r** [comment]
> But it won't be evaluated as JS unless you put it in backticks?

This string is inserted in the generated code as-is - it should not be evaluated in the compile time.

**Comment by valscion** [comment]
Oh wow, that wasn't obvious at all from the webpack documentation. Thanks for explaining.
